kong/db/strategies/cassandra/connector.lua

```diff
@@ -284,44 +279,22 @@ function CassandraConnector:truncate()
       local cql = string.format("TRUNCATE TABLE %s.%s",
                                 self.keyspace, table_name)
 
-      local ok, err = self:query(cql)
+      local ok, err = self:query(cql, nil, nil, "write")
       if not ok then
-        self:setkeepalive()
         return nil, err
       end
     end
   end
 
-  ok, err = self:setkeepalive()
-  if not ok then
-    return nil, err
-  end
```

Comment:
In `:truncate()`, this is done on purpose (sorry if my previous comment wasn't clear enough about that). Because we loop over a significant number of rows, we avoid multiple, unnecessary, connect/keepalive operations. Especially useful in contexts when cosockets are unavailable, and `:setkeepalive()` is closing the underlying LuaSocket connection.

Reply:
but it calls `self:query` in the loop, which does `setkeepalive` that as well, no?

Reply:
Not when the connection is opened prior to calling `:query()`:

https://github.com/Kong/kong/blob/master/kong/db/strategies/cassandra/connector.lua#L196